src/upgrade.ts

```diff
@@ -34,7 +34,6 @@ function upgradeWorker(
     while (needAnotherPass) {
         needAnotherPass = false;
         const program = (lastProgram = createProgramCallback(lastProgram));
-        const checker = program.getTypeChecker();
```

Comment:
I'm prefer to extract the checker in common rather than call the getTypeChecker every time.
Pass type checker and program in parameters is ok.

Reply:
I have checked the implement of the `getTypeChecker` method, it will cache the check internally, so we may not need to keep the check in the closure